src/category_theory/adjunction/fully_faithful.lean

```diff
@@ -63,11 +64,87 @@ instance counit_is_iso_of_R_fully_faithful [full R] [faithful R] : is_iso (adjun
     simp,
   end⟩⟩⟩
 
--- TODO also prove the converses?
--- def L_full_of_unit_is_iso [is_iso (adjunction.unit h)] : full L := sorry
--- def L_faithful_of_unit_is_iso [is_iso (adjunction.unit h)] : faithful L := sorry
--- def R_full_of_counit_is_iso [is_iso (adjunction.counit h)] : full R := sorry
--- def R_faithful_of_counit_is_iso [is_iso (adjunction.counit h)] : faithful R := sorry
+/-- If the unit of an adjunction is an isomorphism, then its inverse on the image of L is given
+by L whiskered with the counit. -/
+@[simp]
+lemma inv_map_unit {X : C} [is_iso (h.unit.app X)] :
+  inv (L.map (h.unit.app X)) = h.counit.app (L.obj X) :=
+is_iso.inv_eq_of_hom_inv_id (h.left_triangle_components)
+
+/-- If the unit of an adjunction is an isomorphism, then its inverse on the image of L is given
+by L whiskered with the counit. -/
+@[simps]
+noncomputable def whisker_left_L_counit_iso_of_is_iso_unit [is_iso (adjunction.unit h)] :
+  L ⋙ R ⋙ L ≅ L :=
+(functor.associator _ _ _).symm ≪≫
+  iso_whisker_right (as_iso (adjunction.unit h)).symm L ≪≫ functor.left_unitor _
+
+/-- If the counit of an adjunction is an isomorphism, then its inverse on the image of R is given
+by R whiskered with the unit. -/
+@[reducible]
+def whisker_left_R_unit_iso_of_is_iso_counit [is_iso (adjunction.counit h)] : R ≅ (R ⋙ L ⋙ R) :=
+⟨ ⟨(whisker_left R h.unit).app, (whisker_left R h.unit).naturality⟩,
+  ⟨(whisker_right h.counit R).app, (whisker_right h.counit R).naturality⟩,
+  begin
+      ext x, dsimp,
+      simp [adjunction.left_triangle, functor.id_obj],
+  end, begin
+      ext x, dsimp,
+      rw [hom_comp_eq_id, ←comp_hom_eq_id],
+      simpa [adjunction.right_triangle, functor.id_obj],
+  end⟩
+
+/-- If the unit is an isomorphism, then the left adjoint is full-/
+noncomputable
+def L_full_of_unit_is_iso [is_iso (adjunction.unit h)] : full L :=
+⟨ λ X Y f, ((h.hom_equiv X (L.obj Y)) f) ≫ (inv (adjunction.unit h)).app Y,
+  λ X Y f,
+  begin
+    simp, symmetry,
+    suffices H : f ≫ L.map (h.unit.app Y) ≫ ((whisker_left_L_counit_iso_of_is_iso_unit h).hom.app Y)
+      = L.map (h.unit.app X) ≫ L.map (R.map f)
+        ≫ ((whisker_left_L_counit_iso_of_is_iso_unit h).hom.app Y),
+    { rw [←assoc, ←assoc] at H,
+      rw [(nat_iso.cancel_nat_iso_hom_right (whisker_left_L_counit_iso_of_is_iso_unit h)
+      (f ≫ L.map (h.unit.app Y)) (L.map (h.unit.app X) ≫ L.map (R.map f)))] at H,
+      rw [←assoc, is_iso.eq_comp_inv],
+      exact H},
+    { simp }
+  end⟩
+
+/-- If the unit is an isomorphism, then the left adjoint is faithful-/
+def L_faithful_of_unit_is_iso [is_iso (adjunction.unit h)] : faithful L :=
+⟨ λ X Y,
+  begin
+    intros f g H,
+    apply_fun (λ k, ((h.hom_equiv X (L.obj Y)) k) ≫ (inv (adjunction.unit h)).app Y) at H,
+    simpa using H,
+  end⟩
```

Comment:
```suggestion
lemma L_faithful_of_unit_is_iso [is_iso (adjunction.unit h)] : faithful L :=
{ map_injective' := λ X Y f g H,
  begin
    rw ←(h.hom_equiv X (L.obj Y)).apply_eq_iff_eq at H,
    simpa using H =≫ inv (h.unit.app Y),
  end }
```
You can slightly simplify this proof, but more importantly `faithful` is a property so it should be a `lemma` rather than a `def`.